### ISSUE TITLE
Setting Max-Age=X on a cookie is completely ignored

### ISSUE THREAD

**Issue by oalders** [libwww-perl/HTTP-Cookies#69]
Reported via email by Robert Mueller.

In this commit <https://github.com/libwww-perl/HTTP-Cookies/commit/dacc2b7b3ba724b0b9932a74fc0edc96a4aa1a0c> the max-age parameter is explicitly singled out to not discard, but unfortunately it introduces a bug because now we skip the main action of:

                else {
                    push(@cur, $k => $v);
                }

Which means it never ends up in `@set`, and thus never ends up in `%hash`, which means:

        my $maxage    = delete $hash{'max-age'};

Is always `undef`.

So basically setting Max-Age=X on a cookie is completely ignored in the current HTTP::Cookies 6.1

Changing:
                elsif (!$first_param && lc($k) eq 'max-age') {
                    $expires++;
                }

To:
                elsif (!$first_param && lc($k) eq 'max-age') {
                    $expires++;
                    push(@cur, "Max-Age" => $v);
                }

Fixes it.

**Comment by oalders** [comment]
Oof. I just got the email about it shortly before the long weekend, so didn't take the time to create a patch and matching test. Was hoping some kind soul might pick this up. 😄

**Comment by wolfsage** [comment]
I wish but this week it ain't me! Short week and vacation next week. Sorry!

**Comment by wolfsage** [comment]
Argh, just bumped into this again

**Comment by oalders** [comment]
I'll have a look today. 😄
